### PR TITLE
FileUploader. Support for custom message body and unauthorized response

### PR DESCRIPTION
**Pull request by masterdany88**
Continuation of:
https://github.com/GwtMaterialDesign/gwt-material-addins/pull/86

Sory for this mess with commits, but I struggled with IDE change.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +0,0 @@
-target/
```
Comment (BenDol) [inline .gitignore]: Need to readd this file

File: .project [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
```
Comment (BenDol) [inline .project]: Add this file to the .gitignore file

File: pom.xml [hunk 3]
```diff
@@ -14,14 +14,14 @@
     <description>Extra Components of GWT Material Framework</description>
```
Comment (BenDol) [inline pom.xml]: Revert the pom changes

File: WebContent/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Manifest-Version: 1.0
```
Comment (BenDol) [inline WebContent/META-INF/MANIFEST.MF]: Remove this path

File: src/main/java/gwt/material/design/addins/client/fileuploader/base/UploadResponse.java [hunk 5]
```diff
@@ -20,34 +20,51 @@
...
 public class UploadResponse {
 
-    private String code;
-    private String message;
+	private String code;
+	private String message;
+	private String body;
+
+	public UploadResponse(String code, String message) {
+		this.code = code;
```
Comment (BenDol) [inline src/main/java/gwt/material/design/addins/client/fileuploader/base/UploadResponse.java]: Replace the tabs with 4 spaces

File: src/main/java/gwt/material/design/addins/client/fileuploader/events/SendingEvent.java [hunk 6]
```diff
@@ -20,54 +20,47 @@
...
-import gwt.material.design.addins.client.fileuploader.base.UploadResponse;
 
 public class SendingEvent<T> extends GwtEvent<SendingEvent.SendingHandler<T>> {
 
-    private static Type<SendingHandler<?>> TYPE;
-
-    public interface SendingHandler<T> extends EventHandler {
-        void onSending(SendingEvent<T> event);
-    }
+	private static Type<SendingHandler<?>> TYPE;
```
Comment (BenDol) [inline src/main/java/gwt/material/design/addins/client/fileuploader/events/SendingEvent.java]: Replace the tabs with 4 spaces

**Comment by masterdany88** [comment]
@BenDol any news about it?

File: src/main/java/gwt/material/design/addins/client/fileuploader/MaterialFileUploader.java [hunk 7]
```diff
@@ -109,35 +134,43 @@ public void add(Widget child) {
     }
 
     public void initDropzone() {
-        String previews = DOM.createUniqueId();
+        final String previews = DOM.createUniqueId();
```
Comment (BenDol) [inline src/main/java/gwt/material/design/addins/client/fileuploader/MaterialFileUploader.java]: Remove all these `final` modifiers, they aren't needed.

File: src/main/java/gwt/material/design/addins/client/fileuploader/MaterialFileUploader.java [hunk 8]
```diff
@@ -109,35 +134,43 @@ public void add(Widget child) {
...
-        if(!isPreview()){
+        if (!isPreview()) {
             uploadPreview.setDisplay(Display.NONE);
         }
-        initDropzone(getElement(), uploadPreview.getUploadCollection().getItem().getElement(), previews, uploadPreview.getElement(),uploadPreview.getUploadHeader().getUploadedFiles().getElement(), getUrl(), getMaxFileSize(), getMaxFiles(), getMethod().getCssName(), isAutoQueue(), getAcceptedFiles(), getClickable(), preview, isWithCredentials());
+        initDropzone(getElement(), uploadPreview.getUploadCollection().getItem().getElement(), previews, uploadPreview.getElement(), uploadPreview.getUploadHeader().getUploadedFiles().getElement(),
+                getUrl(), getMaxFileSize(), getMaxFiles(), getMethod().getCssName(), isAutoQueue(), getAcceptedFiles(), getClickable(), preview, isWithCredentials());
     }
 
+    public static native void alert(String msg) /*-{
```
Comment (BenDol) [inline src/main/java/gwt/material/design/addins/client/fileuploader/MaterialFileUploader.java]: Remove this method.

**Comment by BenDol** [comment]
I can't accept this pull request until the changes I have requested are applied and the commits are [squished](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html). Once that is done I will be able to merge. Thanks

**Comment by masterdany88** [comment]
Thanks for suggestions. I will make this pr better.

**Pull request closed without merging**